community/blog/2023-02 Integrating Arcus Security in a non-invasive manner/README.md

```diff
@@ -0,0 +1,134 @@
+
+After this package is installed, we can initialize the secret store with environment variables and retrieve the Azure Blob storage account key from it:
+
+```csharp
+using Arcus.Security.Core;
+using Azure.Storage;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+var host = new HostBuilder()
+    .ConfigureFunctionsWorkerDefaults()
+    .ConfigureAppConfiguration(builder =>
+    {
+        builder.AddEnvironmentVariables();
+    })
+    .ConfigureSecretStore((config, stores) =>
+    {
+        stores.AddEnvironmentVariables();
+    })
+    .ConfigureServices((context, services) =>
+    {
+        services.AddAzureClients(clients =>
+        {
+            string serviceUrl = context.Configuration["AZURE_BLOB_URL"];
+            string accountName = context.Configuration["AZURE_BLOB_ACCOUNTNAME"];
+
+            clients.AddClient<BlobServiceClient, BlobClientOptions>((options, serviceProvider) =>
+            {
+                var secretProvider = serviceProvider.GetRequiredService<ISecretProvider>();
+                string accountKey = secretProvider.GetRawSecret("AZURE_BLOB_ACCOUNTKEY");
+
+                return new BlobServiceClient(new Uri(serviceUrl), new StorageSharedKeyCredential(accountName, accountKey));
+            });
+        });
+    })
+    .Build();
+
+host.Run();
+```
+
+Notice that the Azure Functions HTTP trigger does not have to change, it is only the retrieval of secrets that has to be adapted. At this point, one may wonder why we initialize both the application configuration and the secret store with the same source (environment variables). And that is a good thing! This means that we start to think of our input as different kinds of types. Both the Arcus environment secret provider and the Microsoft environment variable configuration source can retrieve only a portion of the environment variables by specifying a prefix. This could be a great way to already make the distinction right from the start between application configuration and sensitive secrets.
+
+## Conclusion
+This post showed a way to integrate the Arcus secret store into an existing application without being evasive of the actual application code. It is by design that the example uses Azure Blob storage, as this is an Azure technology we do not necessarily work upon in Arcus (like we do with Service Bus or Event Hubs). It is proof that Arcus can be used in any number of scenarios and should not be limited by the technologies described in our feature documentation.
+
+This example shows how we can separate application configuration from sensitive secrets. This is an important step in making our application more secure and safer to use during future development. Treating secrets as secrets is a mentality shift that will bring great value as it will force you to think about your inputs differently. Not all inputs are public data and should be treated accordingly.
```

Comment:
I think it would be good to mention that you should store your secrets in a secure place as well.  For instance in KeyVault.  Maybe mention that in a next iteration this sample could be modified to retrieve secrets from keyvault instead of retrieving them from env vars.

Because, IMO, it is the goal to securily store secrets, in such way that people cannot just retrieve them.

Reply:
Yes, I agree. 👍
As stated, to be as 'less invasive' as possible, this is already a good first step, but we should mention it definitely, indeed.

Reply:
Added this to the conclusion as 'next step.

Reply:
Indeed, this blogposts shows the first iteration in progressing into a 'better' solution.  Next stop could indeed be that you move secrets to KeyVault and adapt the code to use a KeyVault SecretProvider